### ISSUE TITLE
[A11y]: Checkbox label violation

### ISSUE THREAD

**Issue by davidmenendez** [carbon-design-system/carbon-ai-chat#1334]
Placeholder for an issue that needs to be addressed in the core web component checkbox

<img width="1840" height="1106" alt="Image" src="https://github.com/user-attachments/assets/cafb7269-bd55-4efa-a689-fec1dd86526a" />

This violation shows up in numerous places in our end, so it needs to be addressed

<img width="1840" height="1105" alt="Image" src="https://github.com/user-attachments/assets/7690d135-61cb-4ea4-ab30-9b689779e916" />

**Update**

After some brief, initial investigation: it looks like the issue stems from the usage of `slot` with the checkbox component. using `label-text` instead of the default slot resolves this issue. I'm not sure if this is a general issue with slots or is a false positive from the accessibility checker, but for the time being i'll update our usage to switch from the slot to the prop to alleviate the violations.
